src/cli.rs

```diff
@@ -0,0 +1,821 @@
+#![deny(missing_docs)]
+use crate::client::QuerySecrets;
+use crate::error::NysmError;
+use bat::PrettyPrinter;
+use clap::ValueEnum;
+use clap::{Args, Parser, Subcommand};
+use serde::{Deserialize, Serialize};
+use std::io::IsTerminal;
+use tempdir::TempDir;
+
+/// This struct defines the main command line interface for Nysm.
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+#[command(propagate_version = true)]
+pub struct ArgumentParser {
+  /// Which subcommand to use
+  #[command(subcommand)]
+  pub command: Commands,
+  /// Region to retreive secrets from
+  #[arg(short, long, global = true)]
+  pub region: Option<String>,
+}
+
+/// This enum defines the main command line subcommands for Nysm.
+#[derive(Subcommand, PartialEq, Debug)]
+pub enum Commands {
+  /// Retrieve a list of secrets
+  List(List),
+  /// Edit the value of a specific secret
+  Edit(Edit),
+  /// Show the value of a specific secret
+  Show(Show),
+}
+
+/// Retrieve a list of secrets
+#[derive(Args, PartialEq, Debug)]
+pub struct List {}
```

Comment:
If this is to be implemented in the future, maybe label it as such. 
Something as simple as a comment inside the `List` block saying "TODO".

Reply:
This is implemented. By defining this struct and deriving `args`, clap reads this in as the arguments for the `list` subcommand. Since the list subcommand doesn't accept any arguments, this is fine as is.